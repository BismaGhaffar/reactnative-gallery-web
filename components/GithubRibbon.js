import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Ribbon = styled.img`
  position: absolute;
  top: 0;
  right: 0;
  border: 0;
`

const srcImg =
  'https://s3.amazonaws.com/github/ribbons/forkme_right_white_ffffff.png'

const altTxt = 'Fork me on GitHub'

const GithubRibbon = ({ src, alt }) => (
  <a
    href="http://github.com/ReactNativeGallery/reactnative-gallery-web"
    target="_blank"
    rel="noopener noreferrer"
  >
    <Ribbon src={src} alt={alt} />
  </a>
)

GithubRibbon.propTypes = {
  alt: PropTypes.string,
  src: PropTypes.string
}

GithubRibbon.defaultProps = {
  alt: altTxt,
  src: srcImg
}

export default GithubRibbon
