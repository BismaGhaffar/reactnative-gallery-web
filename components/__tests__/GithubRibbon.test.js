import React from 'react'
import renderer from 'react-test-renderer'
import GithubRibbon from '../GithubRibbon'

it('GithubRibbon can be created', () => {
  const comp = renderer.create(<GithubRibbon />)
  expect(comp).toBeDefined()
})

it('<GithubRibbon /> toMatchSnapshot', () => {
  const tree = renderer.create(<GithubRibbon />).toJSON()
  expect(tree).toMatchSnapshot()
})
