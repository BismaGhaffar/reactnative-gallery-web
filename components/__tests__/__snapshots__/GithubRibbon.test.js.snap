// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GithubRibbon /> toMatchSnapshot 1`] = `
<a
  href="http://github.com/ReactNativeGallery/reactnative-gallery-web"
>
  <img
    alt="Fork me on GitHub"
    className="GithubRibbon__Ribbon-s1o3krv1-0 hvkhtl"
    src="https://s3.amazonaws.com/github/ribbons/forkme_right_white_ffffff.png"
  />
</a>
`;
